include/picongpu/fields/background/templates/twtstight/TWTSTight.tpp

```diff
@@ -0,0 +1,366 @@
+                float_T,
+                float_T,
+                float_T,
+                float_T,
+                float_T,
+                float_T,
+                alpaka::Complex<float_T>,
+                alpaka::Complex<float_T>,
+                float_T,
+                alpaka::Complex<float_T>,
+                alpaka::Complex<float_T>,
+                alpaka::Complex<float_T>>
+            TWTSTight<T_Field>::defineCommonHelperVariables(
+                float_T const& absPhi,
+                float_T const& sinPhi,
+                float_T const& cosPhi,
+                float_T const& beta0,
+                float_T const& tanAlpha,
+                float_T const& cspeed,
+                float_T const& lambda0,
+                float_T const& omega0,
+                float_T const& tauG,
+                float_T const& w0,
+                float_T const& k,
+                float_T const& x,
+                float_T const& y,
+                float_T const& z,
+                float_T const& t,
+                float_T const& cotPhi,
+                float_T const& sinPhi_2) const
+            {
+                using complex_T = alpaka::Complex<float_T>;
+                using complex_64 = alpaka::Complex<float_64>;
+
+                complex_T I = complex_T(0, 1);
+                float_T const x2 = x * x;
+                float_T const tauG2 = tauG * tauG;
+                float_T const psi0 = float_T(2.0) / k;
+                float_T const w02 = w0 * w0;
+                float_T const beta02 = beta0 * beta0;
+
+                float_T const nu = (y * cosPhi - z * sinPhi) / cspeed;
+                float_T const xi = (-z * cosPhi - y * sinPhi) * tanAlpha / cspeed;
+                complex_T const Xm = -z - complex_T(0, 0.5) * (k * w02);
+                complex_T const rhom = math::sqrt(x2 + math::cPow(Xm, static_cast<uint32_t>(2u)));
+                float_T const besselI0const = math::bessel::i0(k * k * sinPhi * w02 / float_T(2.0));
+                complex_T const besselJ0const = math::bessel::j0(k * sinPhi * rhom);
+                complex_T const besselJ1const = math::bessel::j1(k * sinPhi * rhom);
+
+                complex_T const zeroOrder = (beta0 * tauG)
+                    / (math::sqrt(float_T(2.0))
+                       * math::exp(
+                           beta02 * omega0 * math::cPow(t - nu + xi, static_cast<uint32_t>(2u))
+                           / (beta02 * omega0 * tauG2 - complex_T(0, 2) * (beta02 * (nu - xi) * cotPhi * cotPhi)
+                              + complex_T(0, 2) * (beta0 * (float_T(2.0) * nu - xi) * cotPhi / sinPhi)
+                              - complex_T(0, 2) * (nu / sinPhi_2)))
+                       * math::sqrt(
+                           ((beta02 * omega0 * tauG2) / float_T(2.0) - I * (beta02 * (nu - xi) * cotPhi * cotPhi)
+                            + I * (beta0 * (float_T(2.0) * nu - xi) * cotPhi / sinPhi) - I * (nu / sinPhi_2))
+                           / omega0));
```

Comment:
For semantic reasons I fully agree. The question is mainly: Does the compiler generate warnings if I do not use every variable of an unpacked tuple? If yes, then it is not worth un- and repackaging a tailored argument list for zeroOrder if it can just ride here for free.

Reply:
The compiler does not generate warning, that enabled simplifications.